### ISSUE TITLE
Disable tab indentation detection based on extension

### ISSUE THREAD

**Issue by github-account1111** [microsoft/vscode#148608]
`editor.detectIndentation` detects whether to use tabs or spaces for indentation based on the contents of the document - which is nice and what I want.
What isn't so nice is it also seems to decide which indentation char to use based on the _file extension_.
A .yaml file, for instance, will default to spaces if `editor.insertSpaces` is disabled, even in an empty document.
I assume there's some sorta mapping along the lines of "if .java or .js use tabs, if .yaml use spaces," and that mapping ignores whether `editor.insertSpaces` is enabled or not.
I think it either shouldn't ignore it (i.e. if `editor.insertSpaces` is disabled, use tabs in all empty documents) or be configurable.

TLDR I want indentation detected based on contents, not contents _and_ extension.

**Comment by alexdima** [comment]
By default, VS Code configures the following special settings for the `yaml` language:
```
  // Configure settings to be overridden for the yaml language.
  "[yaml]":  {
    "editor.insertSpaces": true,
    "editor.tabSize": 2,
    "editor.autoIndent": "advanced"
  }
```
You can override these in your user settings or workspace settings using the language selector e.g:
```
  // Configure settings to be overridden for the yaml language.
  "[yaml]":  {
    "editor.insertSpaces": false,
    "editor.tabSize": 4,
    "editor.autoIndent": "advanced"
  }
```
---
The upcoming VS Code version will improve the settings UI around language specific settings. In the case of `editor.insertSpaces`, it will now show that, by default, some languages are configured differently:

<img width="1156" alt="image" src="https://user-images.githubusercontent.com/5047891/166882434-b18e2d9f-e4d1-45d2-a463-a48e04535b42.png">
